File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.java
class Solution {
    public int[] sortedSquares(int[] nums) {
        int[] arr = new int[nums.length];
        int start = 0, end = nums.length - 1;
        int index = nums.length - 1;

        while (start <= end) {
            if (Math.abs(nums[start]) > Math.abs(nums[end])) {
                arr[index--] = nums[start] * nums[start];
                start++;
            } else {
                arr[index--] = nums[end] * nums[end];
                end--;
            }
        }
        return arr;
    }
}

File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var sortedSquares = function (nums) {
    let left = 0;
    let right = nums.length - 1;
    let index = nums.length - 1;
    let ans = new Array(nums.length);

    while (left <= right) {
        if (Math.abs(nums[left]) < Math.abs(nums[right])) {
            ans[index--] = nums[right] ** 2;
            right--;
        } else {
            ans[index--] = nums[left] ** 2;
            left++;
        }
    }
    return ans;
};